File: Python/Data Structures & Algorithms Module - Python/Revision/Queues/QueueApp.py
"""
QueueApp.py
Billy Cussen
11/01/2021
"""

from Queue import Queue

myQueue = Queue()

print("Is Empty: "+str(myQueue.isEmpty()))

try:
    print("Size: "+str(myQueue.size()))
except Exception as e:
    print(e)

print(myQueue.toString())

myQueue.enQueue(2)
myQueue.enQueue(4)
myQueue.enQueue(6)
myQueue.enQueue(8)
myQueue.enQueue(10)

print("Is Empty: "+str(myQueue.isEmpty()))
print("Size: "+str(myQueue.size()))
print(myQueue.toString())

myQueue.deQueue()
myQueue.deQueue()
myQueue.deQueue()

print("Is Empty: "+str(myQueue.isEmpty()))
print("Size: "+str(myQueue.size()))
print(myQueue.toString())

File: Java/Data Structures & Algorithms Module - Java/Revision/Queues/QueueApp.java
/*
QueueApp.java
Billy Cussen
11/01/2021
*/

public class QueueApp {
  
    public static void main(String [] args){

        Queues<Integer> myQueue = new Queues<Integer>();

        System.out.println("Is Empty: "+myQueue.isEmpty());
        System.out.println("Size: "+myQueue.size());
        System.out.println(myQueue.toString());

        try{
            System.out.println("Dequeue: "+myQueue.deQueue());
        } catch(NullPointerException e){
            System.out.println(e.getMessage());
        }

        myQueue.enQueue(1);
        myQueue.enQueue(2);
        myQueue.enQueue(4);
        myQueue.enQueue(6);
        myQueue.enQueue(8);
        myQueue.enQueue(10);

        System.out.println("Is Empty: "+myQueue.isEmpty());
        System.out.println("Size: "+myQueue.size());
        System.out.println(myQueue.toString());

        System.out.println("Dequeue: "+myQueue.deQueue());
        System.out.println("Dequeue: "+myQueue.deQueue());
        System.out.println("Dequeue: "+myQueue.deQueue());

        System.out.println(myQueue.toString());


    }
}
